api/envoy/data/dns/v2alpha/dns_table.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+
+package envoy.data.dns.v2alpha;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.data.dns.v2alpha";
+option java_outer_classname = "DnsTableProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: DNS Filter Table Data]
+// :ref:`DNS Filter config overview <config_udp_listener_filters_dns_filter>`.
+
+// This message contains the configuration for the DNS Filter if populated
+// from the control plane
+message DnsTable {
+  // This message contains a list of IP addresses returned for a query for a known name
+  message AddressList {
+    // This field contains a well formed IP address that is returned
+    // in the answer for a name query. The address field can be an
+    // IPv4 or IPv6 address. Address family detection is done automatically
+    // when Envoy parses the string. Since this field is repeated,
+    // Envoy will return one randomly chosen entry from this list in the
+    // DNS response. The random index will vary per query so that we prevent
+    // clients pinning on a single address for a configured domain
+    repeated string address = 1;
+  }
+
+  // This message type is extensible and can contain a list of addresses
+  // or dictate some other method for resolving the addresses for an
+  // endpoint
+  message DnsEndpoint {
+    oneof endpoint_config {
+      AddressList address_list = 1;
+    }
+  }
+
+  message DnsVirtualDomain {
+    // The domain name for which Envoy will respond to query requests
+    string name = 1 [(validate.rules).string = {min_len: 2 well_known_regex: HTTP_HEADER_NAME}];
+
+    // The configuration containing the method to determine the address
+    // of this endpoint
+    DnsEndpoint endpoint = 2;
+
+    // Sets the TTL in dns answers from Envoy returned to the client
+    google.protobuf.Duration answer_ttl = 3 [(validate.rules).duration = {gt {}}];
+  }
+
+  // Control how many times envoy makes an attempt to forward a query to
+  // an external server
+  uint32 external_retry_count = 1;
+
+  // Fully qualified domain names for which Envoy will respond to queries
+  repeated DnsVirtualDomain virtual_domains = 2 [(validate.rules).repeated = {min_items: 1}];
+
+  // This field serves to help Envoy determine whether it can authoritatively
+  // answer a query for a name belonging to a domain in this list. If the query
+  // name has a suffix that does not appear in this list, Envoy will forward
+  // the query to an upstream DNS server
+  repeated string known_domains = 3;
```

Comment:
If this is a suffix list do you want to call is known_domains_suffixes or similar? Or should we use a StringMatcher here? If a suffix list only I assume it could eventually use a trie type match to avoid linear lookup?

Reply:
I had that thought about the trie :)  I can look into StringMatcher and update.

Reply:
Managed to get this going, though I neglected to update the docs in this PR.   I've already made the changes locally and can push them later.

Reply:
Sounds good, go ahead and push and I can do a final review. Thanks!

/wait